Framework/Kernel/inc/MantidKernel/PropertyHelper.h

```diff
@@ -142,6 +144,7 @@ std::string toPrettyString(
   return Strings::shorten(Strings::join(value.begin(), value.end(), delimiter),
                           maxLength);
 }
+GCC_DIAG_ON(unused-function)
```

Comment:
`clang-format` will mess this up! You need to use clang-format off/on, [e.g.](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/inc/MantidPythonInterface/kernel/Converters/NumpyFunctions.h#L32-L34)